pandas/core/indexes/datetimelike.py

```diff
@@ -106,6 +103,21 @@ def freq(self, value) -> None:
     def asi8(self) -> npt.NDArray[np.int64]:
         return self._data.asi8
 
+    @property
+    @doc(DatetimeLikeArrayMixin.freqstr)
+    def freqstr(self) -> str | None:
+        return self._data.freqstr
+
+    @cache_readonly
+    @abstractmethod
+    def _resolution_obj(self) -> Resolution:
```

Comment:
Should theoretically be implemented here but all sub-classes implement it anyways (see https://github.com/pandas-dev/pandas/pull/48015#discussion_r962156351).